File: engine/model/src/main/scala/server/maps/MapleFoothold.scala
package server.maps

import java.awt.Point

class MapleFoothold(private var _firstPoint: Point, private var _secondPoint: Point, private var _id: Int) extends Comparable[MapleFoothold] {
  def id: Int = _id

  def firstX: Int = _firstPoint.x

  def firstY: Int = _firstPoint.y

  def secondX: Int = _secondPoint.x

  def secondY: Int = _secondPoint.y

  def isWall: Boolean = _firstPoint.x == _secondPoint.x

  override def compareTo(o: MapleFoothold): Int = {
    if (secondY < o.firstY) {
      -1
    } else if (firstY > o.secondY) {
      1
    } else {
      0
    }
  }
}

File: engine/engine-model/src/main/java/server/maps/MapleFoothold.java
package server.maps;

import java.awt.Point;

public record MapleFoothold(Point firstPoint, Point secondPoint, Integer id) implements Comparable<MapleFoothold> {
   public Boolean isWall() {
      return firstPoint.getX() == secondPoint.getX();
   }

   @Override
   public int compareTo(MapleFoothold o) {
      if (secondPoint.getY() < o.firstPoint().getY()) {
         return -1;
      } else if (firstPoint.getY() > o.secondPoint().getY()) {
         return 1;
      }
      return 0;
   }
}
